streamlit/pages/3_🤖_Run_Auto_Evaluations.py

```diff
@@ -1,153 +1,243 @@
-import os
-import re
-import streamlit as st
-import psycopg2
-from dotenv import load_dotenv
-from jinja_funcs import *
-import mlflow
+""" 
+Streamlit page for running evaluations in the AutoEval app.
+    
+Functionality:
+- Allows running evaluations on a dataset using selected prompts.
+- Results are stored in the database and can be viewed in the
+    Visualise Results page.
+"""
+
 import pandas as pd
+from dotenv import load_dotenv
+import streamlit as st
+
+from utils import (
+    clear_all_caches, get_prompts, get_samples, get_teachers,
+    generate_experiment_placeholders, start_experiment
+)
+
+load_dotenv()
+
 
 # Set page configuration
 st.set_page_config(page_title="Run Auto Evaluations", page_icon="🤖")
 
-# Function to clear cache
-def clear_all_caches():
-    st.cache_data.clear()
-    st.cache_resource.clear()
-
 # Add a button to the sidebar to clear cache
 if st.sidebar.button('Clear Cache'):
     clear_all_caches()
     st.sidebar.success('Cache cleared!')
 
 # Page and sidebar headers
 st.markdown("# 🤖 Run Auto Evaluations")
-st.sidebar.header("Run Auto Evaluations")
-st.write('This page allows you to run evaluations on a dataset using a selected prompts. Results will be stored in the database and can be viewed in the Visualise Results page.')
-load_dotenv()
+st.write(
+    """
+    This page allows you to run evaluations on a dataset using a
+    selected prompt. Results will be stored in the database and can be 
+    viewed in the Visualise Results page.
+    """
+)
 
 # Initialize session state
-if 'llm_model' not in st.session_state:
-    st.session_state['llm_model'] = 'gpt-4'
+if 'llm_model' not in st.session_state: 
+    st.session_state.llm_model = 'gpt-4'
 if 'llm_model_temp' not in st.session_state:
-    st.session_state['llm_model_temp'] = 0.5
+    st.session_state.llm_model_temp = 0.5
 if 'limit' not in st.session_state:
-    st.session_state['limit'] = 5
+    st.session_state.limit = 5
 if 'created_by' not in st.session_state:
-    st.session_state['created_by'] = 'Select a teacher'
+    st.session_state.created_by = 'Select a teacher'
```

Comment:
We should define this as a constant, as it's duplicated multiple times

Reply:
There's similar instances flagged in Sonar

Reply:
Good to go.